craft_application/services/testing.py

```diff
@@ -48,17 +50,26 @@ def process_spread_yaml(self) -> None:
             craft_backend=craft_backend,
         )
 
-        spread_yaml.to_yaml_file(pathlib.Path("spread/spread.yaml"))
+        spread_yaml.to_yaml_file(dest)
+
+    def run_spread(self, project_path: pathlib.Path) -> None:
+        """Run spread on the processed project file.
+
+        :param project_path: The processed project file.
+        """
+        with emit.pause():
+            os.environ["SPREAD_PROJECT_FILE"] = str(project_path)
+            subprocess.run(["spread", "-v", "craft:"], check=True)
```

Comment:
You may want to use `env=` for passing environment variable for the child process only, additionally 
```suggestion
            os.environ["SPREAD_PROJECT_FILE"] = str(project_path)
            subprocess.check_call(["spread", "-v", "craft:"])
```

Reply:
This will need to be changed later to allow the user to override backends and specify test suites to run.

Reply:
`call`, `check_call`, etc. are just older-api wrappers around `run` - I prefer using `run`.